ikabot/function/distributeResources.py

```diff
@@ -158,99 +171,104 @@ def distribute_evenly(s, resource):
-	totalEspacioDisponible = sum( espacios_disponibles )
-	restanteAEnviar = min(recursoTotal, totalEspacioDisponible)
+	resources_per_city = resources_total // len(destination_cities)
+	spaces_available = [ destination_cities[city]['disponible'] for city in destination_cities ]
+	total_spaces_available = sum( spaces_available )
+	remaining_resources = min(resources_total, total_spaces_available)
 	toSend = {}
 
-	while restanteAEnviar > 0:
+	while remaining_resources > 0:
 		len_prev = len(toSend)
-		for city in ciudadesDestino:
-			ciudad = ciudadesDestino[city]
-			if city not in toSend and ciudad['disponible'] < recursoXciudad:
-				toSend[city] = ciudad['disponible']
-				restanteAEnviar -= ciudad['disponible']
+		for city in destination_cities:
+			city = destination_cities[city]
+			if city not in toSend and city['disponible'] < resources_per_city:
+				toSend[city] = city['disponible']
+				remaining_resources -= city['disponible']
 
 		if len(toSend) == len_prev:
-			for city in ciudadesDestino:
+			for city in destination_cities:
 				if city not in toSend:
-					toSend[city] = recursoXciudad
+					toSend[city] = resources_per_city
 			break
 
-		espacios_disponibles = [ ciudadesDestino[city]['disponible'] for city in ciudadesDestino if city not in toSend ]
-		totalEspacioDisponible = sum( espacios_disponibles )
-		restanteAEnviar = min(restanteAEnviar, totalEspacioDisponible)
-		recursoXciudad = restanteAEnviar // len(espacios_disponibles)
+		spaces_available = [ destination_cities[city]['disponible'] for city in destination_cities if city not in toSend ]
+		total_spaces_available = sum( spaces_available )
+		remaining_resources = min(remaining_resources, total_spaces_available)
+		resources_per_city = remaining_resources // len(spaces_available)
 
 
-	rutas = []
-	for idCiudad in ciudadesDestino:
-		ciudadD = ciudadesDestino[idCiudad]
-		idIsla = ciudadD['islandId']
-		faltante = toSend[idCiudad]
-		for idCiudadOrigen in ciudadesOrigen:
-			if faltante == 0:
+	routes = []
+	for city_id in destination_cities:
+		city_id = destination_cities[city_id]
+		island_id = city_id['islandId']
+		missing_resources = toSend[city_id]
+		for origin_city_id in origin_cities:
+			if missing_resources == 0:
 				break
 
-			ciudadO = ciudadesOrigen[idCiudadOrigen]
-			recursoDisponible = ciudadO['disponible']
-			for ruta in rutas:
-				origen = ruta[0]
-				rec = ruta[recurso + 3]
-				if origen['id'] == idCiudadOrigen:
-					recursoDisponible -= rec
-
-			enviar = min(faltante, recursoDisponible)
-			disponible = ciudadD['disponible']
-			if disponible == 0 or enviar == 0:
+			origin_city = origin_cities[origin_city_id]
+			resources_available = origin_city['disponible']
+			for route in routes:
+				origin = route[0]
+				resource = route[resource_type + 3]
+				if origin['id'] == origin_city_id:
+					resources_available -= resource
+
+			send_this_round = min(missing_resources, resources_available)
+			available = city_id['disponible']
```

Comment:
doesnt look like city id, but real city info
also `available` what? improve naming

Reply:
yeah, it was a city id until line 242, at the start of the loop, should be renamed